autoreject/tests/test_ransac.py

```diff
@@ -50,12 +53,146 @@ def test_ransac():
+                            eog=False, exclude=[])
     ransac = Ransac(picks=picks)
     pytest.raises(ValueError, ransac.fit, epochs)
 
     # should not contain other channel types.
     picks = mne.pick_types(raw.info, meg=False, eeg=True, stim=True,
-                           eog=False, exclude=[])
+                            eog=False, exclude=[])
     ransac = Ransac(picks=picks)
     pytest.raises(ValueError, ransac.fit, epochs)
+
+
+        
+def test_ransac_multiprocessing():
+    """test on real data with 5 random channels augmented with strong
+    50 Hz line noise."""
+    data_path = sample.data_path()
+    raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+    raw = io.read_raw_fif(raw_fname, preload=False)
+
+    raw.del_proj()
+
+    event_id = {'Visual/Left': 3}
+    tmin, tmax = -0.2, 0.5
+
+    events = mne.find_events(raw)
+    epochs = mne.Epochs(raw, events, event_id, tmin, tmax,
+                        baseline=(None, 0), reject=None, preload=True)
+    # see if results are consistent when using different n_jobs
+    # amplify some of the channels with noise as sanity check
+    np.random.seed(0)
+    
+    # create some artifically noisy channels with strong line noise
+    mag_idx = mne.pick_types(epochs.info, meg='mag')
+    noisy_idx = np.random.choice(mag_idx, 5, replace=False)
+    noisy_idx = sorted(noisy_idx)
+    
+    times = np.arange(0, epochs.tmax-epochs.tmin + 1/epochs.info['sfreq'], 
+                      1/epochs.info['sfreq'])
+    sinewave = 10 * np.sin(2*np.pi*50*times)
+        
+    epochs._data[:,noisy_idx,:] *= sinewave
+   
+    
+    # atomic tests for _get_mappings to check if they return the same
+    # result when run "parallel" for subsets or all-in-once, sanity check
+    ransac = Ransac(picks=mag_idx, random_state=np.random.RandomState(42),
+                        n_jobs=1, n_resample=2, min_channels=0.1)
+    ch_subset = ransac._get_random_subsets(epochs.info)
+    ransac.ch_type='meg'
+    mappings = ransac._get_mappings(epochs, ch_subset)
+    mappings_sub1 = ransac._get_mappings(epochs, [ch_subset[0]])
+    mappings_sub2 = ransac._get_mappings(epochs, [ch_subset[1]])
+    
+    np.testing.assert_array_equal(mappings, np.concatenate([mappings_sub1,
+                                                          mappings_sub2]))
+    # atomic tests for _iterate_epochs to check if they return the same
+    # result when run "parallel" for subsets or all-in-once, sanity check
+    ransac.mappings_ = mappings
+    epoch_idxs_splits = np.array_split(np.arange(len(epochs)), 2)
+    corr = ransac._iterate_epochs(epochs, np.arange(len(epochs)))
+    corr_sub1 = ransac._iterate_epochs(epochs, epoch_idxs_splits[0])
+    corr_sub2 = ransac._iterate_epochs(epochs, epoch_idxs_splits[1])
+    np.testing.assert_array_equal(corr, np.concatenate([corr_sub1,
+                                                        corr_sub2]))
+
+    # now test across different jobs
+    mappings = {}
```

Comment:
```suggestion
    mappings = dict()
```

just to be a bit more explicit and different from `list()` or `[]`